### DISCUSSION TITLE
Secure Local Network Access to Krita AI Diffusion / ComfyUI Server

### DISCUSSION THREAD

**Discussion by rbfbatx** [Acly/krita-ai-diffusion#1319] [Ideas]
I've set up an AI + Krita workstation on my desktop PC, but I also want to securely access the Comfy server from my notebook or windows tablet, which also has Krita and AI tools installed. My goal is to leverage the desktop’s resources on the notebook, rather than relying on a tool like Interstice (which has been fantastic, by the way).

Is there an existing solution that allows for this kind of access while prioritizing security? Ideally, I'd want this connection to include an authentication mechanism to prevent unauthorized access to my desktop or, worse.

If this feature doesn’t already exist, I think it would be a great addition for users who work across multiple devices but want to centralize generation/processing on a single machine. Thoughts or suggestions on this?

![laptop in lap of man on couch 01](https://github.com/user-attachments/assets/dd6c1191-baa3-461d-9da5-6bc807eab4f1)

**Comment by Acly** [comment]
For local network this is not hard to do. You start ComfyUI with `--listen` and connect to your desktops IP-address from your Laptop.

If you're using the managed install, you cann pass it like this: https://github.com/Acly/krita-ai-diffusion/wiki/Common-Issues#how-do-i-pass-additional-arguments-to-comfyui

No need to worry about security as it will not be accessible from outside if you're using a private network (standard home LAN).

If you want to access it securely from the internet there are ways, but you better know what you're doing.

**Reply by rbfbatx** [reply]
Thank you!

So I just add "--listen"?

**Reply by minsky91** [reply]
--listen 0.0.0.0

I use a setup very much like yours, already for a few months, except it's not just ComfyUI on my PC I am connecting from the laptop to but a suite of SD servers (some are instances of the same webUI), all managed by Stability Matrix. In general, it works well, but since there is no policy in SM or Windows of intelligently sharing the single CPU to speak of, you have to take care of non-concurrent acces to the GPU yourself, which can be annoying. (meaning, stopping the server that was using GPU last before activating another one - I have to do it remotely via TeamViewer).

**Reply by rbfbatx** [reply]
I've attempted to add the 'listen' argument, both as `--listen` and `--listen 0.0.0.0`, when starting the ComfyUI server on my desktop.

However, when I try to connect to the server from Krita running on my laptop, the connection continuously times out and fails, with a message indicating that the connection has been refused sometimes. I'm unsure of the issue, but could it be that I can't have two instances of Krita running on the server simultaneously? If this is the case, how can I start the server without starting Krita? I would appreciate any suggestions or insights you might have.

**Reply by Acly** [reply]
You can have multiple connections. Probably you just didn't use the correct address.

**Comment by minsky91** [comment]
And indeed, just inserting --llisten won't do, whether with 0.0.0.0 or without, I can confirm that (connection from the Krita AI instance on my laptop would time out every time). What did the trick was adding --port 8190 option to the commandline arguments. After restarting Krita & plugin on the main PC, there was a reaction from Windows which asked to confirm that the network request by the Python app was legitimate, after which I was able to connect to & use the plugin's Comfy server from my laptop.

**Comment by aziniazovaryslanaryslan-debug** [comment]
Action: Tung Tung Sahur riding scooter at 80–90 km/h on a city street
Style: Exaggerated cartoon, funny & slightly creepy expression
Background: Busy city street, colorful, clear sky
Focus: Scooter and Tung Tung Sahur, dynamic motion lines
